pkg/build/controller/buildimagechangetrigger/buildimagechangetrigger_controller.go

```diff
@@ -0,0 +1,287 @@
+				continue
+			}
+			fromStreamName, tag, ok := imageapi.SplitImageStreamTag(from.Name)
+			if !ok {
+				glog.Errorf("Invalid image stream tag: %s in build config %s/%s", from.Name, config.Name, config.Namespace)
+				continue
+			}
+
+			fromNamespace := from.Namespace
+			if len(fromNamespace) == 0 {
+				fromNamespace = config.Namespace
+			}
+
+			// only trigger a build if this image stream matches the name and namespace of the stream ref in the build trigger
+			// also do not trigger if the imagerepo does not have a valid DockerImageRepository value for us to pull
+			// the image from
+			if len(stream.Status.DockerImageRepository) == 0 || fromStreamName != stream.Name || fromNamespace != stream.Namespace {
+				continue
+			}
+
+			latestReference, ok = imageapi.ResolveLatestTaggedImage(stream, tag)
+			if !ok {
+				glog.V(4).Infof("unable to find tagged image: no image recorded for %s/%s:%s", stream.Namespace, stream.Name, tag)
+				continue
+			}
+			glog.V(4).Infof("Found ImageStream %s/%s with tag %s", stream.Namespace, stream.Name, tag)
+
+			// (must be different) to trigger a build
+			last := trigger.ImageChange.LastTriggeredImageID
+			next := latestReference
+
+			if len(last) == 0 || (len(next) > 0 && next != last) {
+				triggeredImage = next
+				shouldBuild = true
+				// it doesn't really make sense to have multiple image change triggers any more,
+				// so just exit the loop now
+				break
+			}
+		}
+
+		if shouldBuild {
+			glog.V(4).Infof("Running build for BuildConfig %s/%s", config.Namespace, config.Name)
+			buildCauses := []buildapi.BuildTriggerCause{}
+			// instantiate new build
+			request := &buildapi.BuildRequest{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      config.Name,
+					Namespace: config.Namespace,
+				},
+				TriggeredBy: append(buildCauses,
+					buildapi.BuildTriggerCause{
+						Message: buildapi.BuildTriggerCauseImageMsg,
+						ImageChangeBuild: &buildapi.ImageChangeCause{
+							ImageID: latestReference,
+							FromRef: from,
+						},
+					},
+				),
+				TriggeredByImage: &kapi.ObjectReference{
+					Kind: "DockerImage",
+					Name: triggeredImage,
+				},
+				From: from,
+			}
+			if _, err := bc.bcInstantiator.Instantiate(config.Namespace, request); err != nil {
+				if kerrors.IsConflict(err) {
+					utilruntime.HandleError(fmt.Errorf("unable to instantiate Build for BuildConfig %s/%s due to a conflicting update: %v", config.Namespace, config.Name, err))
+				} else {
+					utilruntime.HandleError(fmt.Errorf("error instantiating Build from BuildConfig %s/%s: %#v", config.Namespace, config.Name, err))
+				}
+				hasError = true
+				continue
+			}
+		}
+	}
+	if hasError {
+		return fmt.Errorf("an error occurred processing 1 or more build configurations; the image change trigger for image stream/namespace %s/%s and docker image repository %s will be retried", stream.Name, stream.Namespace, stream.Status.DockerImageRepository)
```

Comment:
s/1 or more/one or more/
should modify the message to just say "ImageStream (or whatever we decide should be the right way to say it) %s/%s" with the namespace first instead of "image stream/namespace %s/%s"